logstash-core/lib/logstash/inputs/metrics.rb

```diff
@@ -35,19 +36,22 @@ def stop
     def update(snapshot)
       @logger.debug("Metrics input: received a new snapshot", :created_at => snapshot.created_at, :snapshot => snapshot, :event => snapshot.metric_store.to_event) if @logger.debug?
 
-      # TODO: (ph)
-      # - Obviously the format here is wrong and we need to
-      # transform it from Snapshot to an event
-      # - There is another problem, if the queue is full this could block the snapshot thread.
-      # There is a few possible solution for this problem:
-      #   - We can use a future
-      #   - We can use a synchronization mechanism between the called thread (update method)
-      #   and the plugin thread (run method)
-      #   - How we handle back pressure here?
-      #   - one snashot should be only one event ?
+      # The back pressure is handled in the collector's
+      # scheduled task (running into his own thread) if something append to one of the listener it will
+      # will timeout. In a sane pipeline, with a low traffic of events it shouldn't be a problems.
       snapshot.metric_store.each do |metric|
-        @queue << LogStash::Event.new({ "@timestamp" => snapshot.created_at }.merge(metric.to_hash))
+        @queue << LogStash::Event.new({ "@timestamp" => snapshot.created_at }.merge(convert_to_hash(metric)))
```

Comment:
Oh this is subtile, I found what problem I had before, Jrjackson does the serializing a bit different than the default `json` library, I think this is for performance reason @guyboertje ?

```ruby
require "json"
require "jrjackson"

class MyTest
  attr_reader :message

  def initialize(message)
    @message = message
  end

  def to_hash
    { "message" => "bye world" }
  end

  def to_json(opts = {})
    message
  end

  def to_json_data
    "to_json_data:  #{message}"
  end
end

class MyTest2
  attr_reader :message

  def initialize(message)
    @message = message
  end

  def to_hash
    { "message" => "bye world" }
  end

  def to_json(opts = {})
    message
  end
end

m = MyTest.new("Hello PH")
m2 = MyTest2.new("Hello PH 2")

puts JSON.dump(m)
puts JrJackson::Json.dump(m)
puts JrJackson::Base.generate(m, {})
puts JrJackson::Base.generate(m2, {})
```

```
Hello PH
Hello PH
"to_json_data:  Hello PH"
{"message":"bye world"}
```

If I override `to_json_data` instead of `to_json` it wont call `to_hash`, @guyboertje is there any specific reason why `JrJackson::Json.dump` and `JrJackson::Base.generate` dont call the same methods?